### PR TITLE
Demo issuer callback

### PR DESCRIPTION
**Pull request by sczembor**
+ add callback to demo.issuer `sbt_mint` method
+ fix the bug in oracle `sbt_mint_callback`

### CODE REVIEW

File: contracts/oracle/src/lib.rs [hunk 1]
```diff
@@ -165,16 +165,16 @@ impl Contract {
...
         &mut self,
         external_id: &Vec<u8>,
-        #[callback_result] last_result: Result<TokenId, near_sdk::PromiseError>,
-    ) -> Result<TokenId, near_sdk::PromiseError> {
-        if last_result.is_err() {
-            self.used_identities.remove(&external_id);
+        #[callback_result] last_result: Result<TokenId, PromiseError>,
+    ) -> Option<TokenId> {
+        if last_result.is_ok() {
+            self.used_identities.insert(&external_id);
```
Comment (robert-zaremba) [inline contracts/oracle/src/lib.rs:174]: this is wrong, it was already inserted. We need to remove (rollback) if mint in registry was not succesful.

File: contracts/demo-issuer/src/lib.rs [hunk 2]
```diff
@@ -78,14 +78,25 @@ impl Contract {
...
         }
     }
 
+    #[private]
+    pub fn sbt_mint_callback(
+        &mut self,
+        #[callback_result] last_result: Result<Vec<TokenId>, near_sdk::PromiseError>,
+    ) -> Option<Vec<TokenId>> {
+        if last_result.is_err() {
+            env::panic_str("ERR_CALL_FAILED")
```
Comment (robert-zaremba) [inline contracts/demo-issuer/src/lib.rs:95]:
```suggestion
            env::panic_str("ERR_CALL_FAILED")
```
```suggestion
            env::panic_str("ERR: Mint failed")
```

**Review by robert-zaremba — CHANGES_REQUESTED** [review]
there is a bug in oracle/src/lib.rs

File: contracts/oracle/src/lib.rs (hunk 1 above)
Comment (sczembor) [reply contracts/oracle/src/lib.rs]: Done

File: contracts/demo-issuer/src/lib.rs [hunk 3]
```diff
@@ -78,14 +78,25 @@ impl Contract {
...
         if let Some(memo) = memo {
             env::log_str(&format!("SBT mint memo: {}", memo));
         }
     }
 
+    #[private]
+    pub fn sbt_mint_callback(
+        &mut self,
+        #[callback_result] last_result: Result<Vec<TokenId>, near_sdk::PromiseError>,
+    ) -> Option<Vec<TokenId>> {
```
Comment (robert-zaremba) [inline contracts/demo-issuer/src/lib.rs:93]:
```suggestion
    ) -> Vec<TokenId> {
```

Comment (robert-zaremba) [reply contracts/demo-issuer/src/lib.rs:93]: Why returning option? It should be `Vec`. `last_result` is `Result<Vec<TokenId>, ...>` -- does this PR compile?

**Review by robert-zaremba — APPROVED** [review]
I've fixed the return type

**Pull request merged**